### PR TITLE
Fire user events

### PR DESCRIPTION
**Pull request by stoyandimov**
I'm proposing this as a solution to https://github.com/AsterNET/AsterNET/issues/24 and https://github.com/AsterNET/AsterNET/issues/96 where user event don't get fired.

### CODE REVIEW

**Review by Deantwo — CHANGES_REQUESTED** [review]
This does not look like it would complie.
- [x] You removed the declaration of `currentEvent` but it is still used.
- [x] And you are declaring `userEventHash` inside an `if`'s `then` statement and then attempting to use it on the next line.

I am also unsure what issue you are trying to fix, the two issues you mentioned are old and closed.

**Comment by stoyandimov** [comment]
Thanks for the quick reply. 
This change becomes more readable when you review [the file](https://github.com/AsterNET/AsterNET/blob/134108aca78225485e79121d9eb3a9e5c4e27c80/Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs#L723) or in a [split view](https://github.com/AsterNET/AsterNET/pull/267/files?diff=split&w=0)
Basically there are 2 `TryGetValue` calls to the registered handlers. The first one declares the `out currentEvent` and it attempts to find it by name. If handler doesn't exist, `currentEvent` is `null`. In that case, the second call to `TryGetValue` attempts to find handler for `UserEvent` if that doesn't find a handler, we don't go into the `if` block.

However, if any of two succeed, then the value of `currentEvent` is always a handler. Let me see if I can make it more readable tomorrow.

P.S. My attempt to fix this with as least changes as possible made it unreadable. Apology for that. We are already using it and it is working for sure.

**Comment by Deantwo** [comment]
> This change becomes more readable when you review [the file](https://github.com/AsterNET/AsterNET/blob/134108aca78225485e79121d9eb3a9e5c4e27c80/Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs#L723) or in a [split view](https://github.com/AsterNET/AsterNET/pull/267/files?diff=split&w=0)

You are right, I didn't see it was an `out` there and that the first `if` was removed.
I should just avoid reviewing code on my phone during lunch breaks.

I have no way of testing this, but looking at it you definitely made it neat.

**Review by Deantwo — APPROVED** [review]
Looks good. I just have no idea how `UserEvent` works, so I can't test it.

I assume you tested this much better than I read it at first?

**Comment by stoyandimov** [comment]
> You are right, I didn't see it was an `out` there and that the first `if` was removed. I should just avoid reviewing code on my phone during lunch breaks.

No worries, I guess it comes with maintaining open source projects.

**Comment by stoyandimov** [comment]
**TL;DR**
Without this commit, I couldn't get user event handlers to get fired while normal asterisk event handlers got fired as expected.

With this commit, asterisk event handlers and user event handlers get fired without any errors (in Diagnostics log).

**What happened:**

I'm trying to get a custom event `CustomQueueEvent` to get fired when asterisk emit it.
I followed the explanation of [UserEvent here](https://github.com/AsterNET/AsterNET/blob/master/Asterisk.2013/Asterisk.NET/Manager/Event/UserEvent.cs#L7) and got to the following:

```csharp
var conn = new ManagerConnection("ip", "user", "pass");
// Where asterisk event is UserEvent: CustomQueue
conn.RegisterUserEventClass(typeof(CustomQueueEvent));
conn.DeviceStateChanged += (sender, e) => handler.HandleEvent(e);
conn.UserEvents += (sender, e) => handler.HandleUserEvent(e);
conn.Login();
```

`DeviceStateChanged` gets fired as expected. The only way to get the `CustomQueueEvent` is to set `conn.FireAllEvents = true` and then handle `conn.UnhandledEvent += (_, e) => {...}` ([as one of the issues in my comment suggests as work around](https://github.com/AsterNET/AsterNET/issues/96#issuecomment-293510964)).

I used the master branch of the lib in my application and debugged it. It turned out that `event.Name.GetHashCode()` as resolution to event handlers in `internalEventHandler`, `forgets` that custom events have a different [naming convention](https://github.com/AsterNET/AsterNET/blob/master/Asterisk.2013/Asterisk.NET/Helper.cs#L851).

This PR checks that different naming convention as well. I've been running with this commit for 2 days and see no issue. If anything get's reported and I miss it, please mention me and I'll give my best to fix it.
